[Fix #9895] Disable `Naming/InclusiveLanguage` by default

koic opened the pull request:
Fixes #9895 and follow up https://github.com/rubocop/rubocop/pull/9893#issuecomment-870343243.

This PR disables `Naming/InclusiveLanguage` by default because it has an unexpectedly impact for many users who give feedback.

-----------------

Before submitting the PR make sure the following are checked:

* [x] The PR relates to *only* one subject with a clear title and description in grammatically correct, complete sentences.
* [x] Wrote [good commit messages][1].
* [x] Commit message starts with `[Fix #issue-number]` (if the related issue exists).
* [x] Feature branch is up-to-date with `master` (if not - rebase it).
* [x] Squashed related commits together.
* [ ] Added tests.
* [x] Ran `bundle exec rake default`. It executes all tests and runs RuboCop on its own code.
* [x] Added an entry (file) to the [changelog folder](https://github.com/rubocop/rubocop/blob/master/changelog/) named `{change_type}_{change_description}.md` if the new code introduces user-observable changes. See [changelog entry format](https://github.com/rubocop/rubocop/blob/master/CONTRIBUTING.md#changelog-entry-format) for details.

[1]: https://chris.beams.io/posts/git-commit/

bbatsov commented:
Either this or just disabling the checking of string literals, which I think is the real problem. I'm reasonably certain this won't generate much noise for identifier names and file names. 

In hindsight - it doesn't make much sense for a cop focused on naming to check string literals, as their contents are text, not names. I should have thought of this a bit earlier. :D

bbatsov commented:
We might also remove `master` from the defaults, although I doubt anyone names identiers `master`. :D

tjwp commented:
As the author of this cop, I want to endorse that I'm in support of disabling by default. I think this is a useful tool to put in user's hands but don't want to force it or a specific configuration on anyone.

koic commented:
@bbatsov Sure! I agree with your suggestions! I updated this PR to set `CheckStrings: false` and remove `master` from `FlaggedTerms` for `Naming/InclusiveLanguage`. I think these updates can reduce an unexpectedly impact for users much smaller. So, I reverted the cop's status to pending.

dvandersluis commented:
I agree with making the default configuration more permissive if this cop is going to be enabled by default. It'd be good to have an exhaustive example of how to configure the cop in the docs though so that people could, for example, just apply the original configuration if they are so inclined.
